modules/broker-cli-client/src/main/java/io/ballerina/messaging/broker/client/cmd/impl/list/ListBindingCmd.java

```diff
@@ -0,0 +1,132 @@
+        super(rootCommand);
+    }
+
+    @Override
+    public void execute() {
+        if (help) {
+            processHelpLogs();
+            return;
+        }
+
+        if (!queueName.isEmpty() && !exchangeName.isEmpty()) {
+            BrokerClientException exception = new BrokerClientException();
+            exception.addMessage("either one from 'queue' name and 'exchange' name should only be present");
+            throw exception;
+        }
+
+        // todo: remove this check after the implementation
+        if (!queueName.isEmpty()) {
+            BrokerClientException exception = new BrokerClientException();
+            exception.addMessage("listing bindings of a particular queue is not supported yet");
+            throw exception;
+        }
+
+        Configuration configuration = Utils.readConfigurationFile();
+        HttpClient httpClient = new HttpClient(configuration);
+        String urlParamType = !exchangeName.isEmpty() ? Constants.EXCHANGES_URL_PARAM : Constants.QUEUES_URL_PARAM;
+        String urlParamName = !exchangeName.isEmpty() ? exchangeName : queueName;
+
+        // do GET
+        HttpRequest httpRequest = new HttpRequest(urlParamType + urlParamName + Constants.BINDINGS_URL_PARAM);
+        HttpResponse response = httpClient.sendHttpRequest(httpRequest, "GET");
+
+        // handle data processing
+        ResponseFormatter responseFormatter = new TableFormatter();
+
+        if (response.getStatusCode() == HttpURLConnection.HTTP_OK) {
+            if (!exchangeName.isEmpty()) {
+                Binding[] bindings = processExchangeResponse(response.getPayload(), exchangeName);
+                responseFormatter.printBindingsExchange(bindings);
+            }
+        } else {
+            ResponseFormatter.handleErrorResponse(buildResponseMessage(response, BROKER_ERROR_MSG));
+        }
+
+    }
+
+    @Override
+    public void appendUsage(StringBuilder out) {
+        out.append("Usage:\n");
+        out.append("  " + rootCommand + " list binding (--exchange|--queue) [resource-name] [flag]*\n");
+    }
+
+    /**
+     * Parse the response payload.
+     *
+     * @param payload response payload.
+     * @param exchangeName name of the exchange given at the command.
+     * @return array of binding objects.
+     */
+    private Binding[] processExchangeResponse(String payload, String exchangeName) {
+        JsonParser parser = new JsonParser();
+        JsonArray json = parser.parse(payload).getAsJsonArray();
+        List<Binding> bindings = new ArrayList<>();
+
+        json.forEach(jsonElement -> {
+                    String bindingPatten = jsonElement.getAsJsonObject().get(Binding.BINDING_PATTERN).getAsString();
+                    JsonArray internalArray = jsonElement.getAsJsonObject().getAsJsonArray("bindings");
+                    internalArray.forEach(resource -> {
+                        String queueName = resource.getAsJsonObject().get(Binding.QUEUE_NAME).getAsString();
```

Comment:
Maybe we should rename "queueName" which hides the field declared at line 52.

Reply:
will do